From f5731053aada2731673b4b430926663c9f8168bb Mon Sep 17 00:00:00 2001
From: Rohit Visavalia <rohit.visavalia@xilinx.com>
Date: Fri, 6 Mar 2020 06:40:13 -0800
Subject: [PATCH 1261/1852] sound: soc: xilinx: Use unique dai link per device

commit 76c7e0c69e8ffcd074e15fbc48a4f787a480cc7c from
https://github.com/Xilinx/linux-xlnx.git

Define and assign unique dai link for both dp audio devices and
also add stream_name for device.

Signed-off-by: Rohit Visavalia <rohit.visavalia@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xilinx-dp-card.c | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/sound/soc/xilinx/xilinx-dp-card.c b/sound/soc/xilinx/xilinx-dp-card.c
index 7737acda518b..396a87d56394 100644
--- a/sound/soc/xilinx/xilinx-dp-card.c
+++ b/sound/soc/xilinx/xilinx-dp-card.c
@@ -34,7 +34,12 @@ static const struct snd_soc_ops xilinx_dp_ops = {
 	.startup	= xilinx_dp_startup,
 };
 
-SND_SOC_DAILINK_DEFS(xilinx_dp,
+SND_SOC_DAILINK_DEFS(xilinx_dp0,
+		DAILINK_COMP_ARRAY(COMP_CPU("xilinx-dp-snd-codec-dai")),
+		DAILINK_COMP_ARRAY(COMP_CODEC(NULL, "xilinx-dp-snd-codec-dai")),
+		DAILINK_COMP_ARRAY(COMP_PLATFORM(NULL)));
+
+SND_SOC_DAILINK_DEFS(xilinx_dp1,
 		DAILINK_COMP_ARRAY(COMP_CPU("xilinx-dp-snd-codec-dai")),
 		DAILINK_COMP_ARRAY(COMP_CODEC(NULL, "xilinx-dp-snd-codec-dai")),
 		DAILINK_COMP_ARRAY(COMP_PLATFORM(NULL)));
@@ -42,12 +47,14 @@ SND_SOC_DAILINK_DEFS(xilinx_dp,
 static struct snd_soc_dai_link xilinx_dp_dai_links[] = {
 	{
 		.name		= "xilinx-dp0",
-		SND_SOC_DAILINK_REG(xilinx_dp),
+		.stream_name	= "xilinx-dp0",
+		SND_SOC_DAILINK_REG(xilinx_dp0),
 		.ops		= &xilinx_dp_ops,
 	},
 	{
 		.name		= "xilinx-dp1",
-		SND_SOC_DAILINK_REG(xilinx_dp),
+		.stream_name	= "xilinx-dp1",
+		SND_SOC_DAILINK_REG(xilinx_dp1),
 		.ops		= &xilinx_dp_ops,
 	},
 
-- 
2.31.1

